mobile/src/androidTest/java/ch/epfl/sdp/mobile/test/ui/game/ar/ChessSceneTest.kt

```diff
@@ -1,151 +1,90 @@
-    val piece =
-        object : ChessBoardState.Piece {
-          override val color = ChessBoardState.Color.Black
-          override val rank = ChessBoardState.Rank.Pawn
-        }
-    val state = SinglePieceSnapshotChessBoardState(piece)
-
-    val chessScene =
-        ChessScene(rule.activity.applicationContext, rule.activity.lifecycleScope, state.pieces)
-    assertNotEquals(null, chessScene.boardNode)
-  }
+  fun given_chessScene_whenUpdated_then_haveCorrectNumberOfChildren() = runTest {
+    rule.scenario.onActivity {
+      val chessScene = ChessScene(it.applicationContext, it.lifecycleScope, simpleBoard)
 
-  @Test
-  fun given_stateWithOneQueen_when_initChessScene_then_rootIsNotNull() {
-    val piece =
-        object : ChessBoardState.Piece {
-          override val color = ChessBoardState.Color.Black
-          override val rank = ChessBoardState.Rank.Queen
-        }
-    val state = SinglePieceSnapshotChessBoardState(piece)
-
-    val chessScene =
-        ChessScene(rule.activity.applicationContext, rule.activity.lifecycleScope, state.pieces)
-    assertNotEquals(null, chessScene.boardNode)
-  }
+      val iterator = simpleBoard.entries.iterator()
+      val oldMovePiece = iterator.next()
+      val newMovePiece = oldMovePiece.apply { (Position(4, 4) to value) }.toPair()
 
-  @Test
-  fun given_stateWithOneRook_when_initChessScene_then_rootIsNotNull() {
-    val piece =
-        object : ChessBoardState.Piece {
-          override val color = ChessBoardState.Color.Black
-          override val rank = ChessBoardState.Rank.Rook
-        }
-    val state = SinglePieceSnapshotChessBoardState(piece)
-
-    val chessScene =
-        ChessScene(rule.activity.applicationContext, rule.activity.lifecycleScope, state.pieces)
-    assertNotEquals(null, chessScene.boardNode)
-  }
+      val newPiece = (Position(4, 5) to Piece(EnginePiece(White, Rank.Pawn, PieceIdentifier(40))))
+      val newBoard = mapOf(newMovePiece, newPiece)
 
-  @Test
-  fun given_stateWithOneWhitePawn_when_initChessScene_then_rootIsNotNull() {
-    val piece =
-        object : ChessBoardState.Piece {
-          override val color = ChessBoardState.Color.White
-          override val rank = ChessBoardState.Rank.Pawn
-        }
-    val state = SinglePieceSnapshotChessBoardState(piece)
-
-    val chessScene =
-        ChessScene(rule.activity.applicationContext, rule.activity.lifecycleScope, state.pieces)
-    assertNotEquals(null, chessScene.boardNode)
+      launch {
+        yieldWhile { chessScene.boardNode.children.size != simpleBoard.size }
+
+        chessScene.update(newBoard)
+      }
+
+      launch {
+        yieldWhile { chessScene.boardNode.children.size != newBoard.size }
+
+        assertThat(chessScene.boardNode.children.size).isEqualTo(newBoard.size)
+      }
+    }
   }
 }
+
+/**
+ * This function yield until the [predicate] is not satisfied.
+ *
+ * @param predicate If this function return true yield, otherwise the function end
+ */
```

Comment:
[Nitpick]
```suggestion
/**
 * This function yields until the [predicate] is not satisfied.
 *
 * @param predicate if this function returns true yield, otherwise the function ends.
 */
```